org.eclipse.dawnsci.hdf5.model/src/org/eclipse/dawnsci/hdf5/model/internal/HierarchicalDataFileModel.java

```diff
@@ -87,34 +89,54 @@ private Object getAttribute(String path, String attribute) {
 					return null;
 				}
 				if (node.attributes != null) {
-					return node.attributes.get(attribute);
+					Attribute attr = (Attribute) node.attributes.get(attribute);
+					// attribute does not exist
+					if (attr == null) {
+						return null;
+					}
+					Object attValue = attr.getValue();
+					return NexusUtils.extractScalarFromDataset(attValue);
 				}
 			}
 
 			// attributes have not been loaded
-			try (IHierarchicalDataFile reader = getReader.getReader()) {
-				Map<String, Object> map = reader.getAttributeValues(path);
-				if (map == null) {
-					// path does not exist in file
+			try {
+				NexusFile reader = null;
+				//test if file exists
+				try {
+					reader = getReader.getReader();
+				} catch(NexusException ne) {
+					if (ne.getMessage().startsWith("File not found"))
+						// exit
+						return null;
+				}
+				if (!reader.isPathValid(path)) {
+					// if path does not exist
+					cache.put(path, null);
+					return null;
+				}
+				org.eclipse.dawnsci.analysis.api.tree.Node myNode = reader.getNode(path);
+				Iterator<? extends Attribute> attIterator = myNode.getAttributeIterator();
+				if (!attIterator.hasNext()) {
 					cache.put(path, null);
 					return null;
 				} else {
 					// Extract scalars for all entries
-					for (Entry<String, Object> entry : map.entrySet()) {
-						Object extractedValue = HierarchicalDataUtils
-								.extractScalar(entry.getValue());
-						if (extractedValue != null) {
-							entry.setValue(extractedValue);
-						}
-					}
-
 					Node node = cache.get(path);
 					if (node == null) {
 						node = new Node();
 						cache.put(path, node);
 					}
-					node.attributes = map;
-					return map.get(attribute);
+					if (node.attributes == null) {
+						node.attributes = new HashMap<String, Object>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)